python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -111,6 +113,38 @@ def __new__(
         )
 
 
+@whitelist_for_serdes
+class NotABackfillRequest(
```

Comment:
yay naming! This Request object should not reference backfills and instead should be more about how user communicating their intention to update the set of asset partitions

Some proposed names for this:

- Sync (SyncRequest)
- LaunchRequest
- ExecutionRequest
- MaterializeRequest

Reply:
cc @alangenfeld @schrockn @braunjj for what to name this thing

Reply:
I think Sync came up in our last conversation and was generally liked. I'm still a bit concerned about collision with synced/unsynced status, but maybe that's ok, since `Sync`ing an unsynced asset would bring it up to date. 

I also kind of like `MaterializeRequest`, primarily since it squarely situates this API in asset-world and is already a familiar term

Reply:
IMO the problem with `MaterializeRequest` is that it is too specific to materializable assets. We want to think about extending this to asset checks and observable assets as well as we move forward.

Reply:
One thought here is to really lean into the "asset graph" terminology.

`AssetGraphExecutionRequest` for example just says that the system is requesting that execution happen in the asset graph.